tuning/test_libtuner.py

```diff
@@ -81,320 +73,195 @@ def test_collision_handler():
-    )
-    expect_dump_list = [
-        "Benchmarking: some_dir/baseline.vmfb on device 0 "
-        "BM_main/process_time/real_time_median 60.7 ms 13.5 ms 5 items_per_second=16.4733/s",
-        "Benchmarking result of unet_candidate_1.vmfb on deivce 0 is incomplete\n",
-    ]
-    assert dump_list == expect_dump_list, "fail to parse incomplete candidates"
-
-    b1 = "Benchmarking: baseline.vmfb on device 0"
-    s1 = "Benchmarking: unet_candidate_1.vmfb on device 0"
-    grouped_benchmark_results = [[generate_res(b1, 0), generate_res(s1, 0)]]
-    candidate_trackers[1].model_path = "unet_candidate_1.vmfb"
-    dump_list = libtuner.parse_grouped_benchmark_results(
-        path_config, grouped_benchmark_results, candidate_trackers
-    )
-    expect_dump_list = [
-        "Benchmarking result of baseline.vmfb on deivce 0 is incomplete\n",
-        "Benchmarking result of unet_candidate_1.vmfb on deivce 0 is incomplete\n",
-    ]
-    assert (
-        dump_list == expect_dump_list
-    ), "fail to parse incomplete baseline and candidates"
+    result3 = MagicMock(spec=libtuner.TaskResult)
+    result3.result = MagicMock(stdout="0.98")
+    result3.candidate_id = 0
+    result3.device_id = "device1"
+
+    result4 = MagicMock(spec=libtuner.TaskResult)
+    result4.result = MagicMock(stdout="4.13")
+    result4.candidate_id = 0
+    result4.device_id = "device2"
+
+    candidate_results = [result1, result2]
+    baseline_results = [result3, result4]
+
+    # Mock IREEBenchmarkResult to return float value from stdout
+    def mock_get_median_time(self):
+        return float(self.result_str)
+
+    # Mock IREEBenchmarkResult to return specific benchmark times
+    with patch(
+        "libtuner.IREEBenchmarkResult.get_median_time", new=mock_get_median_time
+    ):
+        # Mock generate_display_MBR to return a fixed display string
+        with patch(
+            "libtuner.generate_display_MBR",
+            side_effect=lambda *args, **kwargs: "display_str",
+        ):
+            # Mock handle_error to avoid actual logging during tests
+            with patch("libtuner.handle_error") as mock_handle_error:
+                # Call the function
+                result = libtuner.parse_model_benchmark_results(
+                    candidate_trackers, candidate_results, baseline_results
+                )
+
+                # Verify interactions with candidate_trackers
+                assert tracker1.model_benchmark_time == 1.23
+                assert tracker1.model_benchmark_device_id == "device1"
+                assert tracker1.baseline_benchmark_time == 0.98
+                assert tracker1.calibrated_benchmark_diff == (1.23 - 0.98) / 0.98
+
+                assert tracker2.model_benchmark_time == 4.56
+                assert tracker2.model_benchmark_device_id == "device2"
+                assert tracker2.baseline_benchmark_time == 4.13
+                assert tracker2.calibrated_benchmark_diff == (4.56 - 4.13) / 4.13
```

Comment:
Floating point representation is not guaranteed to be exact, this should probably test for values being 'almost equal' to the constants on the right